packages/nextjs/src/shared/deprecated.ts

```diff
@@ -0,0 +1,63 @@
+// copy of packages/shared/src/utils/deprecated.ts
```

Comment:
`@clerk/shared` package cannot be used in `@clerk/nextjs` server context due to the following error:


<img width="1071" alt="Screenshot 2023-09-21 at 21 32 30" src="https://github.com/clerkinc/javascript/assets/3936408/8c67cdb5-8418-48f3-8a8b-c60c2103028a">

Reply:
I have tested adding a deprecation in `authMiddleware` and `getAuth` and it seems that the tree-shaking works as expected and the above error is fixed!
I have also tried the `deprecated()` helper in `getServerSideProps` helper of a page and there is also no error. it seems that we can use it safely in server context.